src/server/qgsfilterrestorer.cpp

```diff
@@ -50,6 +50,26 @@ void QgsOWSServerFilterRestorer::applyAccessControlLayerFilters( const QgsAccess
     }
   }
 }
+
+void QgsOWSServerFilterRestorer::applyAccessControlLayerFilters( const QgsAccessControl *accessControl, QgsMapLayer *mapLayer )
+{
+  if ( QgsVectorLayer *layer = qobject_cast<QgsVectorLayer *>( mapLayer ) )
+  {
+    QString sql = accessControl->extraSubsetString( layer );
+    if ( !sql.isEmpty() )
+    {
+      if ( !layer->subsetString().isEmpty() )
+      {
+        sql.prepend( " AND " );
```

Comment:
Better make sure both sides of this AND are contained in brackets `(...) AND (...)`